flit_core/flit_core/sdist.py

```diff
@@ -73,7 +73,7 @@ class SdistBuilder:
     """
     def __init__(self, module, metadata, cfgdir, reqs_by_extra, entrypoints,
                  extra_files, data_directory, include_patterns=(), exclude_patterns=()):
-        self.module = module
+        self.module: common.Module = module
```

Comment:
```suggestion
        self.module = module
```
The annotation should be added to the `__init__` parameter at some point. I'd defer that to some time in the future as it would require reformatting the whole lines.

Reply:
I don't think they have same semantics when you add it to `__init__` parameter or `self.attr` assign. But the type annotation is not the point of this PR, if you disagree with it I'll just remove it.

Reply:
> I don't think they have same semantics when you add it to `__init__` parameter or `self.attr` assign.

Strictly speaking you're right. It's just that IMO attribute annotations don't add much value if they are identical as the parameter ones. I know that's one area where type checkers (in particular mypy and pyright) disagree though.

_In my own experience it works good enough with just parameter annotations._